### ISSUE TITLE
Loading AMD modules fails with "Cannot read property 'length' of null(…)" when initiating a load in `setTimeout` 

### ISSUE THREAD

**Issue by lddubeau** [systemjs/systemjs#1389]
This issue is related to #1386 but when I filed the earlier report, I unwittingly managed to simplify the test case to the point of not being able to reproduce the error reported here. Oops! The difference between the code here and the earlier one is the addition of a module `lib/modD.js` which is required from `lib/modB.js`. Both are part of the `out.js` bundle.
### Versions

At the time of submitting the issue:

systemjs: 0.19.35 and 0.19.36
systemjs-builder: 0.15.25 and 0.15.26
### Steps to reproduce the problem
1. Clone this repo: https://github.com/lddubeau/systemjs_issue2
2. `npm install`.
3. `npm run build` will produce `js/out.js`.
4. Open `index.html` in a server. (You cannot just load it from `file://` due to cross-origin restrictions.)

You may need to reload, or play with the time parameter of `setTimeout` to get an error, the problem is a race condition.

The `index.html` contains the following:

```
<html>
  <head>
    <script type="text/javascript" src="node_modules/systemjs/dist/system.src.js"></script>
    <script type="text/javascript" src="system-config.js"></script>
    <script type="text/javascript">
      System.config({
      bundles: {
        "out.js": [ "lib/modB.js"],
      }
      });
      window.require = SystemJS.amdRequire;
      window.define = SystemJS.amdDefine;
      require(["modA"]);
      setTimeout(function () {
        require(["modC", "lib/modB"]);
      }, 10);
    </script>
  </head>
  <body>
  </body>
</html>
```

The `system-config.js` file contains:

```
System.config({
    baseURL: "js",
    packages: {
        '': {
            defaultExtension: "js"
        }
    }
});
```
### Expected behavior

Everything should load without error. Files should be loaded once and only once.
### Actual behavior

Two HTTP requests are made for `modC.js`

Errors on the console:

```
Uncaught (in promise) Error: (SystemJS) Cannot read property 'length' of null(…)
Uncaught (in promise) Error: (SystemJS) Cannot read property 'length' of null(…)
```

By putting breakpoints and stepping through code I've found that this error is triggered by `normalizedDeps` being `null` at [this line](https://github.com/systemjs/systemjs/blob/3d097d90485a3ceaf5751b5b2709ab7606502105/dist/system.src.js#L3086).

**Comment by guybedford** [comment]
I've managed to replicate this in https://github.com/systemjs/systemjs/commit/dd45df696ee943b0691b1a7758248b787489c6f9.

The previous spec had quite a few race conditions I tried to patch, but this one seems to have slipped through.

**Comment by guybedford** [comment]
This is fixed in [0.20.0-alpha.1](https://github.com/systemjs/systemjs/releases/tag/0.20.0-alpha.1), thanks for the test case.

**Comment by guybedford** [comment]
Fixed in 0.20.0.
